security-analysis/src/main/java/com/powsybl/security/Security.java

```diff
@@ -148,17 +176,99 @@ public static String printLimitsViolations(Network network) {
     }
 
     public static String printLimitsViolations(Network network, LimitViolationFilter filter) {
-        return printLimitsViolations(checkLimits(network), filter);
+        return printLimitsViolations(checkLimits(network), network, filter);
     }
 
+    public static String printLimitsViolations(List<LimitViolation> violations, Network network) {
+        return printLimitsViolations(violations, network, LimitViolationFilter.load());
+    }
+
+    /**
+     * @deprecated Use printLimitsViolations(List<LimitViolation>, Network) instead.
+     */
+    @Deprecated
     public static String printLimitsViolations(List<LimitViolation> violations) {
         return printLimitsViolations(violations, LimitViolationFilter.load());
     }
 
+    public static String printLimitsViolations(List<LimitViolation> violations, Network network, LimitViolationFilter filter) {
+        return printLimitsViolations(violations, network, filter, TableFormatterConfig.load());
+    }
+
+    public static String printLimitsViolations(List<LimitViolation> violations, Network network, LimitViolationFilter filter, TableFormatterConfig formatterConfig) {
+        Objects.requireNonNull(violations);
+        Objects.requireNonNull(filter);
+        Objects.requireNonNull(formatterConfig);
+
+        TableFormatterFactory formatterFactory = new AsciiTableFormatterFactory();
+        Writer writer = new StringWriter();
+        List<LimitViolation> filteredViolations = filter.apply(violations, network);
+
+        NumberFormat numberFormat = getFormatter(formatterConfig.getLocale(), 4, 4);
+        NumberFormat percentageFormat = getFormatter(formatterConfig.getLocale(), 2, 2);
+
+        try (TableFormatter formatter = formatterFactory.create(writer,
+                "",
+                formatterConfig,
+                new Column(COUNTRY),
+                new Column(BASE_VOLTAGE)
+                    .setHorizontalAlignment(HorizontalAlignment.RIGHT),
```

Comment:
The country and the base voltage are related to the voltage level: move these columns after the END column

Reply:
done